private/buf/cmd/buf/command/push/push.go

```diff
@@ -204,3 +217,49 @@ func run(
 	}
 	return nil
 }
+
+// manifestBlob returns a canonical manifest in blob form from a bucket.
+func manifestBlob(
+	ctx context.Context,
+	bucket storage.ReadBucket,
+) (*modulev1alpha1.Blob, error) {
+	moduleManifest, err := manifest.NewFromBucket(ctx, bucket)
+	if err != nil {
+		return nil, err
+	}
+	manifestText, err := moduleManifest.MarshalText()
+	if err != nil {
+		return nil, err
+	}
+	return manifest.NewBlobFromReader(bytes.NewReader(manifestText))
+}
+
+// bucketBlobs returns a deduplicated set of blobs for the files in bucket.
+func bucketBlobs(
+	ctx context.Context,
+	bucket storage.ReadBucket,
+) ([]*modulev1alpha1.Blob, error) {
+	var blobs []*modulev1alpha1.Blob
+	haveBlob := make(map[string]struct{})
+	if walkErr := bucket.Walk(ctx, "", func(info storage.ObjectInfo) error {
```

Comment:
Just tested this, it's including everything in the `buf.yaml` directory. We should have an allow-list when pushing blobs, only "pushable" config files and protos? Here's what I got in the pushed manifest blob (along with all the related blobs):

```
shake256:3a44e5c70e36  LICENSE
shake256:221e20f240f8  buf.gen.yaml
shake256:4ee086005bd8  buf.lock
shake256:09bd2aa76a9d  buf.md
shake256:dd042e390dfd  buf.yaml
shake256:e7f2b8af2efd  foopkg/v1/1.proto
shake256:a9c48acdec2a  foopkg/v1/2.proto
shake256:c9e4511deaf2  foopkg/v1/deps.proto
shake256:b1cff4fd0ef9  some.other.file
```

Pretty sure `buf.gen.yaml` and `some.other.file` shouldn't have been pushed.